### ISSUE TITLE
SQLServerBulkCopy consider caching table metadata

### ISSUE THREAD

**Issue by robertonr** [microsoft/mssql-jdbc#1557]
I'm using driver version 9.2.1 and I see the following code when SQLServerBulkCopy is enabled in class SQLServerPreparedStatement line 1986:

`                try (SQLServerStatement stmt = (SQLServerStatement) connection.createStatement(
                        ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY, connection.getHoldability(),
                        stmtColumnEncriptionSetting);
                        SQLServerResultSet rs = stmt
                                .executeQueryInternal("sp_executesql N'SET FMTONLY ON SELECT * FROM "
                                        + Util.escapeSingleQuotes(tableName) + " '")) {`

That executeQueryInternal mades a round-trip to database but its result should be cached for better performance.

**Comment by peterbae** [comment]
Thanks @robertonr, I think this is a valid suggestion and the team will let you know once we have an update.

**Comment by ruchirchoudhry** [comment]
@robertonr 
Yes caching is a good strategy, but a lot needs to be done to make sure the data in the cache is exactly mirroring the data in the meta tables.
Also every update to the Tx tables or to the tuple , needs to evict of invalidate the data in the caches.
In Tx sync and lock needs to be planned to make this seamless to all the clients.

**Comment by Jeffery-Wasty** [comment]
Resolved by #2464.

**Comment by Jeffery-Wasty** [comment]
Mistakenly closed.
